apps/labs/posts/pytorch-tensoriterator-internals.md

```diff
@@ -0,0 +1,274 @@
+---
+title: 'PyTorch TensorIterator Internals'
+published: April 13, 2020
+author: sameer-deshmukh
+description: 'This post is a deep dive into how TensorIterator works, and is an essential part of learning to contribute to the PyTorch codebase since iterations over tensors in the C++ codebase are extremely commonplace. This post is aimed at someone who wants to contribute to PyTorch'
+category: [Machine Learning]
+featuredImage:
+  src: posts/pytorch-tensoriterator-internals/feature.png
+  alt: 'Code snippet on creating a TensorIterator using a default constructor.'
+hero:
+  imageSrc: posts/pytorch-tensoriterator-internals/blog_hero_var1.svg
+  imageAlt: 'An illustration of a dark brown hand holding up a microphone, with some graphical elements highlighting the top of the microphone.'
+---
+
+> The history section of this post is still relevant, but `TensorIterator`'s
+> interface has changed significantly. For an update on the new API, please check
+> out [this new blog
+> post](https://labs.quansight.org/blog/2021/04/pytorch-tensoriterator-internals-update).
```

Comment:
Per my discussion with Gabriel, there is no equivalent to {: .alert .alert-warning} from the old website styling.

We decided on block quotes for now, as there is no built-in alert function or styling in the new website.

Possible long term solutions we disucssed: MDX plugin or custom tailwind styling

Reply:
For future reference, here's the link to the blog post on the old site, to see how this paragraph used to be styled:
https://quansight-labs.netlify.app/blog/2020/04/pytorch-tensoriterator-internals/